### ISSUE TITLE
Provide working example for GSM calculation via Python

### ISSUE THREAD

**Issue by grzanka** [libamtrack/library#195]
See code reported by @jbrage

```
def GSM_test(E_MeV_u, fluence_cm2):
    
    # this part is probably okay
    particle_no = libam.AT_particle_no_from_Z_and_A_single(1, 1) # Z, A
    material_no = libam.AT_material_number_from_name("Water, Liquid")
    rdd_model_no = libam.RDDModels["RDD_Geiss"].value
    a0_m = 10e-9
    rdd_parameter = [a0_m, 0., 0.]
    er_model = libam.AT_ERModels["ER_Edmund"].value
    stopping_power_source_no = libam.stoppingPowerSource_no["PSTAR"].value
    
    # this part is probably wrong ...
    nX = 100 # what is nX?
    pixel_size_m = 1e-6 # too coarse, just a test ..
    number_of_bins = 100 # hmm?
    dose_bin_centers = 10 ** np.linspace(np.log10(0.001), np.log10(1), number_of_bins)
    random_number_generator_seed = [0] * number_of_bins
    zero_dose_fraction = [0] * number_of_bins
    dose_frequency_Gy = [0] * number_of_bins
    
    libam.AT_GSM_local_dose_distrib(p_E_MeV_u=[E_MeV_u], 
                                    p_fluence_cm2=[fluence_cm2], 
                                    p_particle_no=[particle_no],
                                    p_material_no=material_no, 
                                    p_rdd_model=rdd_model_no,
                                    p_rdd_parameter=rdd_parameter,
                                    p_er_model=er_model,
                                    p_stopping_power_source_no=stopping_power_source_no,
                                    p_nX=nX,
                                    p_pixel_size_m=pixel_size_m,
                                    p_number_of_bins=number_of_bins,
                                    p_dose_bin_centers_Gy=dose_bin_centers.tolist(),
                                    p_random_number_generator_seed=random_number_generator_seed,
                                    p_zero_dose_fraction=zero_dose_fraction,
                                    p_dose_frequency_Gy=dose_frequency_Gy,
                                   )
```

**Comment by grzanka** [comment]
Lets check if this will work with following code:
```
import numpy as np
import pyamtrack.libAT as libam

def GSM_test(E_MeV_u, fluence_cm2):

    # this part is probably okay
    particle_no = libam.AT_particle_no_from_Z_and_A_single(1, 1) # Z, A
    material_no = libam.AT_material_number_from_name("Water, Liquid")
    rdd_model_no = libam.RDDModels["RDD_Geiss"].value
    a0_m = 10e-9
    rdd_parameter = [a0_m, 0., 0.]
    er_model = libam.AT_ERModels["ER_Edmund"].value
    stopping_power_source_no = libam.stoppingPowerSource_no["PSTAR"].value

    # this part is probably wrong ...
    nX = 100 # what is nX?
    pixel_size_m = 1e-6 # too coarse, just a test ..
    number_of_bins = 100 # hmm?
    dose_bin_centers = 10 ** np.linspace(np.log10(0.001), np.log10(1), number_of_bins)
    random_number_generator_seed = [2705490069]
    zero_dose_fraction = [0]
    dose_frequency_Gy = [0] * number_of_bins


    libam.AT_GSM_local_dose_distrib(p_E_MeV_u=[E_MeV_u], 
                                    p_fluence_cm2=[fluence_cm2], 
                                    p_particle_no=[particle_no],
                                    p_material_no=material_no, 
                                    p_rdd_model=rdd_model_no,
                                    p_rdd_parameter=rdd_parameter,
                                    p_er_model=er_model,
                                    p_stopping_power_source_no=stopping_power_source_no,
                                    p_nX=nX,
                                    p_pixel_size_m=pixel_size_m,
                                    p_dose_bin_centers_Gy=dose_bin_centers.tolist(),
                                    p_random_number_generator_seed=random_number_generator_seed,
                                    p_zero_dose_fraction=zero_dose_fraction,
                                    p_dose_frequency_Gy=dose_frequency_Gy,
                                   )

    print("zero_dose_fraction = ", zero_dose_fraction)
    print("dose freq", dose_frequency_Gy)

if __name__ == '__main__':
    GSM_test(60., 1e6)
```

**Comment by grzanka** [comment]
Seems to work with pyamtrack v0.14.0:

```
gitpod /workspace/library (grzanka/provide-working-example-195) $ pip install pyamtrack numpy
Collecting pyamtrack
  Downloading pyamtrack-0.14.0-py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (5.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.0/5.0 MB 84.0 MB/s eta 0:00:00
Collecting numpy
  Downloading numpy-1.23.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.1/17.1 MB 120.1 MB/s eta 0:00:00
Requirement already satisfied: cffi>=1.13.0 in /home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from pyamtrack) (1.15.0)
Requirement already satisfied: pycparser in /home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from cffi>=1.13.0->pyamtrack) (2.21)
Installing collected packages: numpy, pyamtrack
Successfully installed numpy-1.23.3 pyamtrack-0.14.0

[notice] A new release of pip available: 22.1.2 -> 22.2.2
[notice] To update, run: pip install --upgrade pip
gitpod /workspace/library (grzanka/provide-working-example-195) $ python issue195.py 
zero_dose_fraction =  [0.0]
dose freq [0.0411, 0.0361, 0.0297, 0.0259, 0.0201, 0.0159, 0.0154, 0.014, 0.0098, 0.0095, 0.0091, 0.008, 0.0064, 0.0055, 0.006, 0.0046, 0.0038, 0.0035, 0.0031, 0.0027, 0.0027, 0.0014, 0.0014, 0.0018, 0.0015, 0.0009, 0.0016, 0.001, 0.0009, 0.0012, 0.0013, 0.0008, 0.0007, 0.0005, 0.0007, 0.0006, 0.0009, 0.0009, 0.0003, 0.0006, 0.0005, 0.0004, 0.0006, 0.0004, 0.0003, 0.0002, 0.0002, 0.0001, 0.0002, 0.0003, 0.0004, 0.0004, 0.0002, 0.0002, 0.0002, 0.0, 0.0001, 0.0001, 0.0001, 0.0001, 0.0, 0.0001, 0.0002, 0.0, 0.0001, 0.0, 0.0, 0.0001, 0.0, 0.0, 0.0, 0.0001, 0.0001, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0001, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
```

Older versions are broken:
```
gitpod /workspace/library (grzanka/provide-working-example-195) $ pip install "pyamtrack==0.13.0"
Collecting pyamtrack==0.13.0
  Downloading pyamtrack-0.13.0-py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (5.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.0/5.0 MB 78.5 MB/s eta 0:00:00
Requirement already satisfied: cffi>=1.13.0 in /home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from pyamtrack==0.13.0) (1.15.0)
Requirement already satisfied: pycparser in /home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from cffi>=1.13.0->pyamtrack==0.13.0) (2.21)
Installing collected packages: pyamtrack
  Attempting uninstall: pyamtrack
    Found existing installation: pyamtrack 0.14.0
    Uninstalling pyamtrack-0.14.0:
      Successfully uninstalled pyamtrack-0.14.0
Successfully installed pyamtrack-0.13.0

[notice] A new release of pip available: 22.1.2 -> 22.2.2
[notice] To update, run: pip install --upgrade pip
gitpod /workspace/library (grzanka/provide-working-example-195) $ python issue195.py 
Traceback (most recent call last):
  File "issue195.py", line 45, in <module>
    GSM_test(60., 1e6)
  File "issue195.py", line 25, in GSM_test
    libam.AT_GSM_local_dose_distrib(p_E_MeV_u=[E_MeV_u], 
TypeError: AT_GSM_local_dose_distrib() missing 1 required positional argument: 'p_number_of_bins'
```
